core/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -613,7 +614,10 @@ private static void declareMetaDataFields(ObjectParser<Map<String, Object>, Void
         Map<String, SearchHits> innerHits = new HashMap<>();
         while ((parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
-            innerHits.put(parser.currentName(), SearchHits.fromXContent(parser));
+            String name = parser.currentName();
+            ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+            ensureFieldName(parser, parser.nextToken(), SearchHits.Fields.HITS);
+            innerHits.put(name, SearchHits.fromXContent(parser));
```

Comment:
This needed to change because tests uncovered that we didn't correctly parse this structure. It looks like this:
```
"inner_hits": {
           "<inner_hits_name>": {
              "**hits**": {
                 "total": ...,
                 "hits": [
                    {
                       "_type": ...,
                       "_id": ...,
                       ...
                    },
                    ...
                 ]
              }
           }
        },
        ...
```
but it turns out we somehow gracefully read over the first `hits` level and parsed the rest correctly, but this doesn't work any more when there are random bits in between.